mlir/lib/Catalyst/Transforms/TBAAPatterns.cpp

```diff
@@ -72,7 +108,7 @@ struct MemrefLoadTBAARewritePattern : public ConvertOpToLLVMPattern<memref::Load
             loadOp.getNontemporal());
 
         if (isAnyOf<IndexType, IntegerType, FloatType, MemRefType>(baseType)) {
```

Comment:
You can add the a function that check that is index and `isFromExtractAlignedPointerAsIndexOp(currentOp) || isInsideDeallocHelper(currentOp)` then you return ptr or index and do not need to change the settag